bezier-rs/lib/src/lib.rs

```diff
@@ -271,53 +306,62 @@ impl Bezier {
 	}
 
 	/// Return an approximation of the length of the bezier curve.
-	pub fn length(&self) -> f64 {
-		// Code example from <https://gamedev.stackexchange.com/questions/5373/moving-ships-between-two-planets-along-a-bezier-missing-some-equations-for-acce/5427#5427>.
-
-		// We will use an approximate approach where
-		// we split the curve into many subdivisions
-		// and calculate the euclidean distance between the two endpoints of the subdivision
-		let lookup_table = self.compute_lookup_table(Some(LENGTH_SUBDIVISIONS));
-		let mut approx_curve_length = 0.0;
-		let mut prev_point = lookup_table[0];
-		// calculate approximate distance between subdivision
-		for curr_point in lookup_table.iter().skip(1) {
-			// calculate distance of subdivision
-			approx_curve_length += (*curr_point - prev_point).length();
-			// update the prev point
-			prev_point = *curr_point;
-		}
+	/// - `num_subdivisions` - Number of subdivisions used to approximate the curve. The default value is 1000.
+	pub fn length(&self, num_subdivisions: Option<i32>) -> f64 {
```

Comment:
I wonder if we could use SIMD for this function?

Reply:
simd is a bit tricky because you always need a reference to the previous iteration.
Processing this in parallel would likely not yield great results due to the extra overhead.
But we can do lots by improving the algorithm used.
Raph Levien (kurbo author) has a great article on exactly that and has gotten into a lot more details + did benchmarks on this: https://raphlinus.github.io/curves/2018/12/28/bezier-arclength.html